docs/src/main/asciidoc/s3.adoc

```diff
@@ -378,6 +378,244 @@ There can be multiple `S3ClientCustomizer` beans present in single application c
+3. Yaml (application.yaml for example)
+
+File resolved with `spring.config.import` can be also referenced in `application.properties`.
+For example, with a file `urlDev.json` containing following JSON structure:
+
+[source,json]
+----
+{
+      "url": "someUrl"
+}
+----
+
+
+`spring.config.import` entry is added to `application.properties`:
+
+[source, properties]
+----
+spring.config.import=aws-s3:/myPath/application.json
+----
+
+File configuration values can be referenced by JSON key names:
+
+[source, java]
+----
+@Value("${url}"
+private String url;
+----
+
+=== Customizing S3Client
+
+To use custom `S3Client` in `spring.config.import`, provide an implementation of `BootstrapRegistryInitializer`. For example:
+
+[source,java]
+----
+package com.app;
+
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistryInitializer;
+
+public class S3ClientBootstrapConfiguration implements BootstrapRegistryInitializer {
+
+    @Override
+    public void initialize(BootstrapRegistry registry) {
+        registry.register(S3Client.class, context -> {
+            AwsCredentialsProvider awsCredentialsProvider = StaticCredentialsProvider.create(AwsBasicCredentials.create("yourAccessKey", "yourSecretKey"));
+            return S3Client.builder().credentialsProvider(awsCredentialsProvider).region(Region.EU_WEST_2).build();
+        });
+    }
+}
+----
+
+Note that this class must be listed under `org.springframework.boot.BootstrapRegistryInitializer` key in `META-INF/spring.factories`:
+
+[source, properties]
+----
+org.springframework.boot.BootstrapRegistryInitializer=com.app.S3ClientBootstrapConfiguration
+----
+
+If you want to use autoconfigured `S3Client` but change underlying SDKClient or ClientOverrideConfiguration you will need to register bean of type `AwsS3ClientCustomizer`:
+Autoconfiguration will configure `S3Client` Bean with provided values after that, for example:
+
+[source,java]
+----
+package com.app;
+
+import io.awspring.cloud.autoconfigure.config.s3.AwsS3ClientCustomizer;
+import java.time.Duration;
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistryInitializer;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.services.s3.S3Client;
+
+class S3ClientBootstrapConfiguration implements BootstrapRegistryInitializer {
+
+	@Override
+	public void initialize(BootstrapRegistry registry) {
+		registry.register(AwsS3ClientCustomizer.class,
```

Comment:
`AwsS3ClientCustomizer` is deprecated. Instead we should show in examples usages of `S3ClientCustomizer`